src/v/iceberg/manifest_entry_type.cc

```diff
@@ -0,0 +1,107 @@
+// Copyright 2024 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+#include "iceberg/manifest_entry_type.h"
+
+namespace iceberg {
+
+namespace {
+
+struct_type data_file_type(partition_key_type partition_type) {
+    struct_type r2_type;
+    // TODO: somehow make this not a magic number.
+    r2_type.fields.reserve(16);
```

Comment:
Can we add a bunch of `static constexpr int32_t`s somewhere for the magic numbers/field IDs?

Reply:
I can, but they would only really be used here, and they shouldn't ever change given they're part of the Iceberg spec. I can do this if you feel strongly about it